flang/lib/Optimizer/Transforms/AffinePromotion.cpp

```
@@ -312,18 +312,29 @@ mlir::Type coordinateArrayElement(fir::ArrayCoorOp op) {
 std::pair<mlir::AffineApplyOp, fir::ConvertOp>
 createAffineOps(mlir::Value arrayRef, mlir::PatternRewriter &rewriter) {
   auto acoOp = arrayRef.getDefiningOp<ArrayCoorOp>();
-  auto genDim = acoOp.dims().getDefiningOp<GenDimsOp>();
+  auto genDim = acoOp.getShape().getDefiningOp<ShapeOp>();
```

we should throw an error in case of slice if not supported.